types/klaw-sync/index.d.ts

```diff
@@ -1,40 +1,62 @@
-// Type definitions for klaw-sync 1.1
+// Type definitions for klaw-sync 3.0
 // Project: https://github.com/manidlou/node-klaw-sync
 // Definitions by: Brendan Forster <https://github.com/shiftkey>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 /// <reference types="node" />
 
 import * as fs from 'fs'
-
 declare namespace klawSync {
-  interface Item {
-    path: string
-    stats: fs.Stats
-  }
-
-  interface Options {
-    /**
-     *  any paths or `micromatch` patterns to ignore.
-     *
-     * For more information on micromatch patterns: https://github.com/jonschlinkert/micromatch#features
-     */
-    ignore?: string | string[]
-    /**
-     * True to only return files (ignore directories).
-     *
-     * Defaults to false if not specified.
-     */
-    nodir?: boolean
-    /**
-     * True to only return directories (ignore files).
-     *
-     * Defaults to false if not specified.
-     */
-    nofile?: boolean
-  }
+    interface Item {
+        path: string
+        stats: fs.Stats
+    }
+
+    type Filter = (item: Item) => boolean
+
+    interface Options {
+        /**
+         * @description any paths or `micromatch` patterns to ignore.
+         *              For more information on micromatch patterns: https://github.com/jonschlinkert/micromatch#features
+         * @deprecated v2.0.0
+         */
+        ignore?: string | string[]
```

Comment:
If this has been deprecated in `2.0` then I'd just remove the parameter so that it's clear what's currently supported.

Reply:
Let's focus on the v2 changes here, it seems like v3 is a smoother upgrade that we can follow up with later.